src/liballoc/borrow.rs

```diff
@@ -207,6 +207,50 @@ impl<B: ?Sized + ToOwned> Clone for Cow<'_, B> {
 }
 
 impl<B: ?Sized + ToOwned> Cow<'_, B> {
+    /// Returns true if the data is borrowed, i.e. if [`to_mut`] would require additional work.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(cow_is_borrowed)]
+    /// use std::borrow::Cow;
+    ///
+    /// let mut cow = Cow::Borrowed("moo");
+    /// assert!(cow.is_borrowed());
+    ///
+    /// let mut bull: Cow<'_, str> = Cow::Owned("...moo?".to_string());
+    /// assert!(!bull.is_borrowed());
+    /// ```
+    #[unstable(feature = "cow_is_borrowed", issue = "65143")]
+    pub fn is_borrowed(&self) -> bool {
+        match *self {
+            Borrowed(_) => true,
+            Owned(_) => false,
+        }
+    }
+
+    /// Returns true if the data is owned, i.e. if [`to_mut`] would be a no-op.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(cow_is_borrowed)]
+    /// use std::borrow::Cow;
+    ///
+    /// let mut cow: Cow<'_, str> = Cow::Owned("moo".to_string());
+    /// assert!(cow.is_owned());
+    ///
+    /// let mut bull = Cow::Borrowed("...moo?".to_string());
+    /// assert!(!bull.is_owned());
+    /// ```
+    #[unstable(feature = "cow_is_borrowed", issue = "65143")]
+    pub fn is_owned(&self) -> bool {
+        match *self {
+            Borrowed(_) => false,
+            Owned(_) => true,
+        }
```

Comment:
Why not: `!self.is_borrowed()`? (see [`is_some`/`is_none`](https://github.com/rust-lang/rust/blob/4ac4809ccf5f77083ae7155dcc83e921341c2614/src/libcore/option.rs#L184-L210), [`is_ok`/`is_err`](https://github.com/rust-lang/rust/blob/4ac4809ccf5f77083ae7155dcc83e921341c2614/src/libcore/result.rs#L283-L310))